Essentials/src/com/earth2me/essentials/utils/LocationUtil.java

```diff
@@ -89,26 +86,42 @@ public static Location getTarget(final LivingEntity entity) throws Exception {
     }
 
     public static boolean isBlockAboveAir(final World world, final int x, final int y, final int z) {
-        return y > world.getMaxHeight() || HOLLOW_MATERIALS.contains(world.getBlockAt(x, y - 1, z).getType());
+        return isBlockAboveAir(world.getChunkAt(x, z), x, y, z);
+    }
+
+    public static boolean isBlockAboveAir(final Chunk chunk, int x, final int y, int z) {
+        x = x & 0xF;
+        z = z & 0xF;
+        return y > chunk.getWorld().getMaxHeight() || HOLLOW_MATERIALS.contains(chunk.getBlock(x, y - 1, z).getType());
```

Comment:
```suggestion
        return y > chunk.getWorld().getMaxHeight() || HOLLOW_MATERIALS.contains(chunk.getBlock(x & 0xf, y - 1, z & 0xf).getType());
```

While Java doesn't do this, other languages this kind of assignment is scary, as you may be modifying the value out of scope. (You aren't here, you are fine, but it isn't a good pattern)